### ISSUE TITLE
Incorrect values in MOUSEHOOKSTRUCTEX_MOUSE_DATA

### ISSUE THREAD

**Issue by HavenDV** [microsoft/CsWin32#518]
## Actual behavior

```cs
[Flags]
internal enum MOUSEHOOKSTRUCTEX_MOUSE_DATA : uint
{
  XBUTTON1 = 0x00000001,
  XBUTTON2 = 0x00000002,
}
```
The current code assumes use in the form:
```cs
var isXButton1 = value.mouseData.HasFlag(MOUSEHOOKSTRUCTEX_MOUSE_DATA.XBUTTON1);
var isXButton2 = value.mouseData.HasFlag(MOUSEHOOKSTRUCTEX_MOUSE_DATA.XBUTTON2);
```
But that won't work because the values of the constants are wrong.

Code that works for me at the moment:
```cs
var specialKey = (((uint)value.mouseData) >> 16) switch
{
    1 => Key.MouseXButton1,
    2 => Key.MouseXButton2,
    _ => Key.MouseNone,
};
```

## Expected behavior
```cs
var isXButton1 = value.mouseData.HasFlag(MOUSEHOOKSTRUCTEX_MOUSE_DATA.XBUTTON1);
var isXButton2 = value.mouseData.HasFlag(MOUSEHOOKSTRUCTEX_MOUSE_DATA.XBUTTON2);
```
should return correct values

## Repro steps

1. `NativeMethods.txt` content:
```
MSLLHOOKSTRUCT
```

2. `NativeMethods.json` content (if present):
```json
```

3. Any of your own code that should be shared?

### Context

- CsWin32 version: [e.g. `0.1.635-beta`]
- Win32Metadata version (if explicitly set by project):
- Target Framework: [e.g. `netstandard2.0`]
- `LangVersion` (if explicitly set by project): [e.g. `9`]

**Comment by HavenDV** [comment]
I don't think this is a problem with this generator. But maybe having the documentation in `MOUSEHOOKSTRUCTEX_MOUSE_DATA` or the second part of the documentation in `MSLLHOOKSTRUCT.mouseData` would make this problem easier.

**Comment by AArnott** [comment]
The metadata is defining `XBUTTON1` and `XBUTTON2` as 32-bit values, but the docs indicate they are 16-bit values. That's a bug in the metadata. 
Further, given these constants only make up a _part_ of the `mouseData` field's value (the high-order 16-bits), I don't think the field should be typed as an enum with these constants at all. Leave it as a `uint` and let folks do the bit-shifting as they'd have to do in native code as well, but without the enum misleading them into believing the values will actually match.

I'm moving this issue to the metadata repo.
